sympy/functions/elementary/integers.py

```diff
@@ -220,3 +220,72 @@ def __ge__(self, other):
         if self.args[0] == other and other.is_real:
             return S.true
         return Ge(self, other, evaluate=False)
+
+
+class frac(Function):
+    r"""Represents the fractional part of x
+
+    For real numbers it is defined [1]_ as
+
+    .. math::
+        x - \lfloor{x}\rfloor
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, frac, Rational, floor, ceiling, I
+    >>> frac(Rational(4, 3))
+    1/3
+    >>> frac(-Rational(4, 3))
+    2/3
+
+    returns zero for integer arguments
+
+    >>> n = Symbol('n', integer=True)
+    >>> frac(n)
+    0
+
+    rewrite as floor
+
+    >>> x = Symbol('x')
+    >>> frac(x).rewrite(floor)
+    x - floor(x)
+
+    for complex arguments
+
+    >>> r = Symbol('r', real=True)
+    >>> frac(I*r)
+    I*frac(r)
+
+    See Also
+    ========
+
+    sympy.functions.elementary.integers.floor
+    sympy.functions.elementary.integers.ceiling
+
+    References
+    ===========
+
+    .. [1] http://en.wikipedia.org/wiki/Fractional_part
+    .. [2] http://mathworld.wolfram.com/FractionalPart.html
+
+    """
+    @classmethod
+    def eval(cls, arg):
+        from sympy import im
+        if arg.is_integer:
+            return S.Zero
+        if arg.is_imaginary or (S.ImaginaryUnit*arg).is_real:
```

Comment:
I still find it disturbing that only pure imaginary arguments are treated differently. Otherwise the result seems to remain unevaluated with arguments containing symbols.
```
>>> r = Symbol('r', real=True)
>>> frac(r*I)
I*frac(r)
>>> frac(1 + r*I)
frac(I*r + 1)    # instead of I*frac(r)
```